### PR TITLE
Yaml File Generator

### PR DESCRIPTION
**Pull request by mariari**
Currently we have 6~ libraries that have overlapping dependencies. Currenlty it is a pain to change all versions at the same time.

This PR stands to generate those various yaml files while giving a good way to share dependency code and extend our depdencies

### CODE REVIEW

File: scripts/yaml-generator.lisp [hunk 1]
```diff
@@ -0,0 +1,304 @@
...
+  (make-groups :comment "Morley Specific dependencies"
+               :deps (list
+                      *tezos-bake-monitor*)))
+
+
+;; -----------------------------------
+;; stack-yaml for the YAML generation
+;; -----------------------------------
+
+(defparameter *standard-library*
```
Comment (Acentelles) [inline scripts/yaml-generator.lisp:257]: Missing some libraries? (e.g. translate, core, ...)

File: Makefile [hunk 2]
```diff
@@ -73,4 +73,7 @@ clean:
 clean-full:
 	stack clean --full
 
+stack-yaml:
+	ros -Q scripts/yaml-generator.lisp
```
Comment (cyberglot) [inline Makefile:77]: This ensures it works out of the box, right?

File: doc/CONTRIBUTING.md [hunk 3]
```diff
@@ -100,3 +100,16 @@ Please put the following in `.git/hooks/pre-commit` and run `chmod +x .git/hooks
...
+## New Libraries
+
+Juvix works in a rather modular design, with most loosly coupled
+features being a library in the lanugage. One can start a new library
+with `stack new <Projectname>`. This will generate a lot of cruft that
+you'll want to remove to make it look like the other projects.
+
+Please note that we have a special `stack.yaml` generator because
+there was a lot of repeat code. See the file itself for how to mock
+(big comment at the top).
```
Comment (cyberglot) [inline doc/CONTRIBUTING.md:113]: Could you add a link here? 😬

File: scripts/yaml-generator.lisp [hunk 4]
```diff
@@ -0,0 +1,779 @@
...
+
+;; --------------------------------------
+;; Interaction Net Groups Depencenies
+;; --------------------------------------
+
+(defparameter *interaction-net-extra-deps*
+  (make-groups :comment "For Interaction Nets json-schema"
+               :deps (list
+                      (make-dependency-github
+                       :name "cryptiumlabs/jsonschema-gen"
```
Comment (cyberglot) [inline scripts/yaml-generator.lisp:528]: Shouldn't it be `heliaxdev` here?

File: scripts/yaml-generator.lisp [hunk 5]
```diff
@@ -0,0 +1,779 @@
...
+
+;; --------------------------------------
+;; Stadnard Library Style Dependencies
+;; --------------------------------------
+
+(defparameter *capability*
+  (string->dep-sha "capability-0.4.0.0@sha256:d86d85a1691ef0165c77c47ea72eac75c99d21fb82947efe8b2f758991cf1837,3345"))
+
+(defparameter *extensible*
+  (make-dependency-github :name "metastatedev/extensible-data"
```
Comment (cyberglot) [inline scripts/yaml-generator.lisp:399]: `heliaxdev` as well?

**Review by cyberglot — COMMENTED** [review]
I'm a bit worried about hashes being hardcoded as strings, how do we update them if necessary?

**Review by cyberglot — APPROVED** [review]

File: Makefile (hunk 2 above)
Comment (mariari) [reply Makefile:77]: yes, as we all have `ros` on our machines due to the org generator. This just tells it to run it when you say `make stack-yaml`

File: doc/CONTRIBUTING.md (hunk 3 above)
Comment (mariari) [reply doc/CONTRIBUTING.md:113]: I've added a link to the generator file, if that is what you meant.

File: scripts/yaml-generator.lisp (hunk 5 above)
Comment (mariari) [reply scripts/yaml-generator.lisp:399]: Probably, but I was keeping the semantics the same from the old file, thus not changing any dependency

**Comment by Acentelles** [comment]
Should also update https://github.com/heliaxdev/juvix/blob/develop/package.yaml#L2 accordingly

**Comment by Acentelles** [comment]
Do you want to follow the convention of feat, fix, etc?

File: library/EasyPipeline/stack.yaml [hunk 6]
```diff
@@ -6,92 +6,89 @@ packages:
 - ../Frontend
 - ../Core
 - ../Translate
+- ../Backends/Michelson
```
Comment (Acentelles) [inline library/EasyPipeline/stack.yaml:9]: Remove

Comment (mariari) [reply library/EasyPipeline/stack.yaml:9]: It's needed to make it compile

Comment (mariari) [reply library/EasyPipeline/stack.yaml:9]: This was part of the chnages needed to make `EasyPipeline` compile

**Pull request merged**